File: Interpreter/AdditiveExpr.java
package calculator.xwg;

public class AdditiveExpr extends NonterminalExpr {
	static Expr buildExpr(BuildContext context){
		class MyProxy implements ChildExprBuildProxy{
			public Expr buildExpr(BuildContext context){
				return MultiplicativeExpr.buildExpr(context);
			}
		} 
		
		class MyCreater implements ParentCreater {
			public NonterminalExpr newInstance(){
				return new AdditiveExpr();
			}
		}
		return buildExpr(new MyProxy(), new MyCreater(), context, "[-+]");
    }
	public ValueOperator getValueOperator(String operatorContent){
    	class MyOperator extends ValueOperator
        {
	        public MyOperator(String _operatorString){
	    		super(_operatorString);
	    	}
            public boolean evaluate(Complex value1, Complex value2)
            {
                if(operatorString.compareTo("+") == 0)
                {
                    evaluateResult = Complex.add(value1, value2);
                    return true;
                }
                else// if(operatorString.compareTo("-") == 0)
                {
                	evaluateResult = Complex.sub(value1, value2);
                    return true;
                }
            }
        };
        return new MyOperator(operatorContent);
    }
}

File: Interpreter/AdditiveExpr.py
from Interpreter.NonterminalExpr import *
from Interpreter.ValueOperator import *
from Interpreter.MultiplicativeExpr import *


class AdditiveExpr(NonterminalExpr):
    # 构建表达式
    @staticmethod
    def buildExpr(context):
        # 构建加法表达式
        return NonterminalExpr.buildExpr4(context, AdditiveExpr, MultiplicativeExpr, "[-+]")

    def getValueOperator(self, operatorContent):
        class MyOperator(ValueOperator):
            def evaluate(self, value1, value2):
                if self.operatorString == "+":
                    self.evaluateResult = Complex.add(value1, value2)
                    return True
                else: #if (operatorString.compareTo("-") == 0)
                    self.evaluateResult = Complex.sub(value1, value2)
                    return True
        return MyOperator(operatorContent)
